json_validation/repositories/json_repository.py

```
@@ -0,0 +1,202 @@
+                    try:
+                        index = int(part)
+                        current = current[index] if 0 <= index < len(current) else None
+                    except (ValueError, IndexError):
+                        return None
+                else:
+                    return None
+
+                if current is None:
+                    return None
+
+            return current
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)